### PR TITLE
Clean up build/usage requirements for hdMaya and mayaToHydra targets.

### PR DESCRIPTION
**Pull request by HamedSabri-adsk**
This is the first of many small PRs to clean up build/usage requirements logics for targets inside maya-usd/lib.

### CODE REVIEW

File: lib/usd/hdMaya/utils.cpp [hunk 1]
```diff
@@ -14,7 +14,7 @@
 // limitations under the License.
 //
 
-#include <hdMaya/utils.h>
```
Comment (kxl-adsk) [inline lib/usd/hdMaya/utils.cpp]: Utils is not a private header file. @HamedSabri-adsk please revert this change.

File: lib/usd/hdMaya/CMakeLists.txt [hunk 2]
```diff
@@ -56,53 +53,17 @@ target_compile_definitions(${TARGET_NAME}
...
-        ${TBB_INCLUDE_DIRS}
-        ${CMAKE_BINARY_DIR}
 )
 
 # -----------------------------------------------------------------------------
 # link libraries
 # -----------------------------------------------------------------------------
 target_link_libraries(${TARGET_NAME} 
     PUBLIC 
-        arch
```
Comment (kxl-adsk) [inline lib/usd/hdMaya/CMakeLists.txt]: Nice!

File: lib/usd/hdMaya/utils.cpp (hunk 1 above)
Comment (HamedSabri-adsk) [reply lib/usd/hdMaya/utils.cpp]: @kxl-adsk Can you clarify what private header mean here?

We have a utils.h and utils.cpp sitting besides each other. Why do I need to do `hdMaya/utils.h` instead of simply `utils.h`

Besides, I have `${CMAKE_CURRENT_SOURCE_DIR}` added to `target_include_directories`  as `PRIVATE` requirement in `lib/usd/hdMaya/CMakeLists.txt`. 

```
target_include_directories(${TARGET_NAME} 
    PRIVATE
        ${CMAKE_BINARY_DIR}/include
        ${CMAKE_CURRENT_SOURCE_DIR}
)
```

Comment (elrond79) [reply lib/usd/hdMaya/utils.cpp]: Hmm... so I was curious if this was / could be made a private header or not - and I think we may need to it to be public, because any hdMaya adapters that others might wish to implement will need/want it.

Currently, it's effectively public, since it's installed.  However, it's referenced in many places using "../hdMaya/utils.h" syntax, which is slightly confusing, and will need to be fixed to get everything in line with our style conventions.

Comment (HamedSabri-adsk) [reply lib/usd/hdMaya/utils.cpp]: > 
> 
> Hmm... so I was curious if this was / could be made a private header or not - and I think we may need to it to be public, because any hdMaya adapters that others might wish to implement will need/want it.
> 
> Currently, it's effectively public, since it's installed. However, it's referenced in many places using "../hdMaya/utils.h" syntax, which is slightly confusing, and will need to be fixed to get everything in line with our style conventions.

Please see https://github.com/Autodesk/maya-usd/pull/369 and https://github.com/Autodesk/maya-usd/pull/370
